backend/auth/handlers.go

```diff
@@ -0,0 +1,340 @@
+	if err != nil {
+		logger.DbLog.Errorln(err.Error())
+		return nil, errors.New(errorRetrieveUserAccounts)
+	}
+	var users []*configmodels.User
+	users = make([]*configmodels.User, 0)
+	for _, rawUser := range rawUsers {
+		var user configmodels.User
+		err := json.Unmarshal(configmodels.MapToByte(rawUser), &user)
+		if err != nil {
+			logger.DbLog.Errorf(errorRetrieveUserAccount)
+			continue
+		}
+		user.Password = ""
+		users = append(users, &user)
+	}
+	return users, nil
+}
+
+func GetUserAccount(c *gin.Context) {
+	logger.WebUILog.Infoln("get user account")
+	username := c.Param("username")
+	user, err := fetchUser(username)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccount})
+		return
+	}
+	if user == nil {
+		c.JSON(http.StatusNotFound, gin.H{"error": errorUsernameNotFound})
+		return
+	}
+	user.Password = ""
+	c.JSON(http.StatusOK, user)
+}
+
+func fetchUser(username string) (*configmodels.User, error) {
+	filter := bson.M{"username": username}
+	rawUser, err := dbadapter.WebuiDBClient.RestfulAPIGetOne(UserAccountDataColl, filter)
+	if err != nil {
+		logger.DbLog.Errorln(err.Error())
+		return nil, err
+	}
+	if len(rawUser) == 0 {
+		return nil, nil
+	}
+	var user configmodels.User
+	err = json.Unmarshal(configmodels.MapToByte(rawUser), &user)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		return nil, err
+	}
+	return &user, nil
+}
+
+func PostUserAccount(c *gin.Context) {
+	logger.WebUILog.Infoln("create user account")
+	var user configmodels.User
+	err := c.ShouldBindJSON(&user)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidDataProvided})
+		return
+	}
+	if user.Username == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingUsername})
+		return
+	}
+	shouldGeneratePassword := user.Password == ""
+	if shouldGeneratePassword {
+		generatedPassword, passwordErr := generatePassword()
+		if passwordErr != nil {
+			logger.AuthLog.Errorln(passwordErr.Error())
+			c.JSON(http.StatusInternalServerError, gin.H{"error": errorCreateUserAccount})
+			return
+		}
+		user.Password = generatedPassword
+	}
+	if !validatePassword(user.Password) {
+		logger.AuthLog.Errorln("invalid password provided")
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidPassword})
+		return
+	}
+	user.Permissions = USER_ACCOUNT
+	isFirstAccountIssued, err := IsFirstAccountIssued()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccounts})
+		return
+	}
+	if !isFirstAccountIssued {
+		user.Permissions = ADMIN_ACCOUNT
+	}
+	password := user.Password
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.Password), bcrypt.DefaultCost)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorCreateUserAccount})
+		return
+	}
+	user.Password = string(hashedPassword)
```

Comment:
This manipulation is quite ugly, we should not use the same attribute to represent both the actual password and the hashed password. 

The db object should be something like

```go
type DBUser struct {
	id             int64
	username       string
	hashedPassowrd string
	permissions    int
}
```

and the API User should be 

```go
type CreateUserParams struct {
	Username    string `json:"username"`
	Password    string `json:"password"`
	Permissions int    `json:"permissions"`
}
```